package com.skillstorm.week1.day4;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;
import java.util.List;

class MyClass implements AutoCloseable {

	@Override
	public void close() throws Exception {
		// TODO Auto-generated method stub
		// Sever connection a database
	}
	
}

public class FileIOPractice {

	public static void writeToFile(String filePath, String message) {
		// BufferedWriter
		
		// try-with-resources
		// If the try finishes or an exception occurs, the file is autoclosed
		try (BufferedWriter writer = new BufferedWriter(new FileWriter(filePath, true))) {
			writer.write(message);
		} catch (IOException e) {
			// prompt the user to provide a different file
			e.printStackTrace();
		}
		
		
		// Old way with no try-with-resources
//		BufferedWriter writer = null;
//		try {
//			writer = new BufferedWriter(new FileWriter(filePath));
//			writer.write(message);
//			
//			// Some logic
//			
//			
//		} catch (IOException e) {
//			// TODO Auto-generated catch block
//		} finally { // This block "always" runs
//			try {
//				writer.close();
//			} catch (IOException e) {
//				// TODO Auto-generated catch block
//				e.printStackTrace();
//			} catch (NullPointerException e) {
//				
//			}
//		}
	}
	
	public static List<String> getFileContents(String filePath) {
		try (BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
			
		} catch (FileNotFoundException e) {
			// Prompt the user to choose a different file/display some error
			e.printStackTrace();
		} catch (IOException e) { // The closing of the file could cause an IOException
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
		
		return null;
	}
	
	public static void main(String[] args) {
		// IO
		// IO stands Input/Output
		
		// Relative pathing
		writeToFile("./files/myFile2.txt", "Hello World!\nAuf Wiedersehen!\n");
//		writeToFile("myFile.txt", "");
	}

}
